File: sample-java/src/test/java/sample/banana/java/ThirdTest.java
package sample.banana.java;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.lang.reflect.Field;

import static org.junit.jupiter.api.Assertions.assertSame;

final class ThirdTest {
  private First.Second.Third<String> subject;

  @BeforeEach
  void setUp() {
    subject = new First.Second().new Third<>();
  }

  @Test
  void setAField() throws NoSuchFieldException, IllegalAccessException {
    final String expected = "this is a mock value";

    FirstSecondThirdTestAccessors.setYetAnotherField(subject, expected);

    final Field field = subject.getClass().getDeclaredField("yetAnotherField");
    final boolean wasAccessible = field.isAccessible();
    field.setAccessible(true);
    final String actual = (String) field.get(subject);
    assertSame(expected, actual);
    field.setAccessible(wasAccessible);
  }
}

File: sample-kotlin/src/test/kotlin/sample/banana/kotlin/ThirdTest.kt
package sample.banana.kotlin

import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import sample.banana.kotlin.FirstSecondThirdTestAccessors.yetAnotherField

internal class ThirdTest {
	private lateinit var subject: First.Second.Third<String>

	@BeforeEach
	fun setUp() {
		subject = First.Second().Third()
	}

	@Test
	fun setAField() {
		val expected = "this is a mock value"

		subject.yetAnotherField(expected)

		var actual: String?
		subject::class.java.getDeclaredField("yetAnotherField").apply {
			val wasAccessible = isAccessible
			isAccessible = true
			actual = this[subject] as String?
			isAccessible = wasAccessible
		}
		assertSame(expected, actual)
	}
}
